tests/engine/hammer_test.go

```diff
@@ -12,61 +12,21 @@ import (
 	"github.com/tetratelabs/wazero/wasm"
 )
 
-func TestJITConcurrency(t *testing.T) {
+var hammers = map[string]func(t *testing.T, r wazero.Runtime){
+	// Tests here are similar to what's described in /RATIONALE.md, but deviate as they involve blocking functions.
+	"close importing module while in use": closeImportingModuleWhileInUse,
+	"close imported module while in use":  closeImportedModuleWhileInUse,
+}
+
+func TestEngineJIT_hammer(t *testing.T) {
 	if !wazero.JITSupported {
 		t.Skip()
 	}
-	runAdhocTestsUnderHighConcurrency(t, wazero.NewRuntimeConfigJIT)
-	singleModuleHighConcurrency(t, wazero.NewRuntimeConfigJIT)
-}
-
-func TestInterpreterConcurrency(t *testing.T) {
-	runAdhocTestsUnderHighConcurrency(t, wazero.NewRuntimeConfigInterpreter)
-	singleModuleHighConcurrency(t, wazero.NewRuntimeConfigInterpreter)
+	runAllTests(t, hammers, wazero.NewRuntimeConfigJIT())
 }
 
-func runAdhocTestsUnderHighConcurrency(t *testing.T, newRuntimeConfig func() *wazero.RuntimeConfig) {
-	t.Run("huge stack", func(t *testing.T) {
-		t.Parallel()
-		runAdhocTestUnderHighConcurrency(t, newRuntimeConfig, testHugeStack)
```

Comment:
ps these didn't really do what we thought they would as each uses a separate runtime :D It would be a lot of refactoring to make each "ad-hoc" test work with a shared runtime, particularly around dodging module names (ex most tests use the same module name "host". Instead, I removed these from the concurrency aka hammer tests as they aren't valid and distract from what's actually testing concurrency vs just running the same test many times (like a benchmark)